### PR TITLE
extract #createInjector

### PR DESCRIPTION
**Pull request by jschneider**
This will help me a lot.

I am embedding Jooby into a Guice based Swing application.
Being able to create a child injector will be of great help.

### CODE REVIEW

**Comment by jknack** [comment]
I prefer to do as much as possible with override...

can you add an `.injector(BiFunction<Stage, Module> provider)`?

example:

```java
Jooby app = new Jooby();
app.injector((stage, module) -> Guice.createInjector(stage, module));
```

but also:

```java
app.injector(Guice::createInjector);
```

**Comment by jschneider** [comment]
Okay, I updated that method. Also created two test cases.

File: coverage-report/src/test/java/org/jooby/issues/PullRequest583.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import com.google.inject.Injector;
+import org.jooby.test.ServerFeature;
+import org.junit.*;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
+ */
+public class PullRequest583 extends ServerFeature {
```
Comment (jknack) [inline coverage-report/src/test/java/org/jooby/issues/PullRequest583.java]: This won't run from maven. Rename your test as `Issue583` or append `Test`, like `PullRequest583Test`

File: coverage-report/src/test/java/org/jooby/issues/PullRequest583WithChildInjector.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.google.inject.Injector;
+import org.jooby.test.ServerFeature;
+import org.junit.*;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * @author Johannes Schneider (<a href="mailto:js@cedarsoft.com">js@cedarsoft.com</a>)
+ */
+public class PullRequest583WithChildInjector extends ServerFeature {
```
Comment (jknack) [inline coverage-report/src/test/java/org/jooby/issues/PullRequest583WithChildInjector.java]: Same comment here.

File: coverage-report/src/test/java/org/jooby/issues/PullRequest583WithChildInjector.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  private final AtomicReference<Injector> createdInjector = new AtomicReference<>();
+
+  private final Injector parentInjector = Guice.createInjector(
+    binder -> binder.bind(MyInjectedClass.class).toInstance(new MyInjectedClass()));
+
+  {
+
+    injector((stage, module) -> {
+      Injector injector = parentInjector.createChildInjector(module);
```
Comment (jknack) [inline coverage-report/src/test/java/org/jooby/issues/PullRequest583WithChildInjector.java]: 👍

File: jooby/src/main/java/org/jooby/Jooby.java [hunk 4]
```diff
@@ -2727,6 +2744,18 @@ private Injector bootstrap(final Config args,
     return injector;
   }
 
+  /**
+   * Creates the injector using the given module.
+   * Call {@link #injector(BiFunction)} to configure how the injector is created.
+   */
+  protected Injector createInjector(Stage stage, com.google.inject.Module joobyModule) {
```
Comment (jknack) [inline jooby/src/main/java/org/jooby/Jooby.java]: remove this method there is always an injectorFactory

**Comment by jknack** [comment]
Thanks @jschneider  Looks good, few more comments

**Comment by jknack** [comment]
Also, try to -amend your commits, so we generated just 1 commit per pull request.

File: coverage-report/src/test/java/org/jooby/issues/PullRequest583WithChildInjector.java (hunk 2 above)
Comment (jschneider) [inline coverage-report/src/test/java/org/jooby/issues/PullRequest583WithChildInjector.java]: done

File: jooby/src/main/java/org/jooby/Jooby.java (hunk 4 above)
Comment (jschneider) [inline jooby/src/main/java/org/jooby/Jooby.java]: done

**Comment by jschneider** [comment]
Ok. I start to understand your code style ;-)
Please take a look again.

**Comment by jknack** [comment]
Thanks! will wait for travis and then merge.

**Pull request merged**